# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: servo/components/style/properties/longhands/box.mako.rs
# Commit: 473ebe5ec7aa
# Full Hash: 473ebe5ec7aa4f1c5da45ed580fb56258737a210
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-08-23 04:21:33
# Regressor Bug: 1819464
# File Overlap Count: 2
# Description:
#   Bug 1819464 - Part 2: Add content-box and stroke-box into transform-box in style. r=emilio
#   
#   Also, add a prefernce for content-box and stroke-box. So we have to
#   define a new type, TransformBox, and let content-box and stroke-box
#   behind the preference.
# ==============================================================================

diff -r 9c22e16e0567 -r 473ebe5ec7aa servo/components/style/properties/longhands/box.mako.rs
--- a/servo/components/style/properties/longhands/box.mako.rs	Tue Aug 22 19:49:03 2023 +0000
+++ b/servo/components/style/properties/longhands/box.mako.rs	Tue Aug 22 19:49:03 2023 +0000
@@ -447,13 +447,12 @@
     affects="paint",
 )}
 
-${helpers.single_keyword(
+${helpers.predefined_type(
     "transform-box",
-    "view-box border-box fill-box",
+    "TransformBox",
+    "computed::TransformBox::ViewBox",
     engines="gecko",
-    gecko_enum_prefix="StyleGeometryBox",
     spec="https://drafts.csswg.org/css-transforms/#transform-box",
-    gecko_inexhaustive="True",
     animation_value_type="discrete",
     affects="overflow",
 )}